# DSI-06 Homework 1 (Python) ANSWERS

### 8. This question involves the use of simple linear regression on the Auto data set.

#### Import libraries and specific objects

In [35]:
import numpy as np
import pandas as pd
from matplotlib.pyplot import subplots
import statsmodels.api as sm

from ISLP import load_data
from ISLP.models import (ModelSpec as MS,
                         summarize,
                         poly)

#### Load 'Auto' data set

In [13]:
Auto = load_data("Auto")
Auto.columns # to see associated variables

Index(['mpg', 'cylinders', 'displacement', 'horsepower', 'weight',
       'acceleration', 'year', 'origin', 'name'],
      dtype='object')

In [43]:
print(Auto) # return column names and first few rows of the data set

      mpg  cylinders  displacement  horsepower  weight  acceleration  year  \
0    18.0          8         307.0         130    3504          12.0    70   
1    15.0          8         350.0         165    3693          11.5    70   
2    18.0          8         318.0         150    3436          11.0    70   
3    16.0          8         304.0         150    3433          12.0    70   
4    17.0          8         302.0         140    3449          10.5    70   
..    ...        ...           ...         ...     ...           ...   ...   
387  27.0          4         140.0          86    2790          15.6    82   
388  44.0          4          97.0          52    2130          24.6    82   
389  32.0          4         135.0          84    2295          11.6    82   
390  28.0          4         120.0          79    2625          18.6    82   
391  31.0          4         119.0          82    2720          19.4    82   

     origin                       name  
0         1  chevrolet

#### a)	Use the *sm.OLS()* function to perform a simple linear regression with *mpg* as the response and *horsepower* as the predictor. Use the *summarize()* function to print the results. Comment on the output. For example:

i. Is there a relationship between the predictor and the response? 

ii. How strong is the relationship between the predictor and the response?

iii. Is the relationship between the predictor and the response positive or negative?

iv. What is the predicted mpg associated with a horsepower of 98? What are the associated 95 % confidence and prediction intervals?


In [18]:
Y = Auto['mpg']         # Response variable
X = Auto['horsepower']  # Predictor variable

In [46]:
# Use the lm() function to fit our linear regression: lm(y ~ x)
# Extract results and fit model
y = Auto['mpg']
model = sm.OLS(y, X)
results = model.fit()

In [48]:
# Use the 'print' function to see full model summary
print(Auto_model.summary())

                                 OLS Regression Results                                
Dep. Variable:                    mpg   R-squared (uncentered):                   0.649
Model:                            OLS   Adj. R-squared (uncentered):              0.648
Method:                 Least Squares   F-statistic:                              723.7
Date:                Sat, 28 Oct 2023   Prob (F-statistic):                    5.67e-91
Time:                        20:04:43   Log-Likelihood:                         -1608.1
No. Observations:                 392   AIC:                                      3218.
Df Residuals:                     391   BIC:                                      3222.
Df Model:                           1                                                  
Covariance Type:            nonrobust                                                  
                 coef    std err          t      P>|t|      [0.025      0.975]
-----------------------------------------

In [41]:
# Summarize model results using the summarize() function
summarize(Auto_model)

coef  std err       t  P>|t|
horsepower  0.1788    0.007  26.901    0.0

In [50]:
# The fitted coefficients can be retrieved using the 'params' attribute of results
Auto_model.params

horsepower    0.17884
dtype: float64

i) So we have the coefficient estimates, their associated standard errors, and the t-statistic and p-value
associated with the hypothesis test H0 : 1 = 0. The p-value for this test is significant so we can
conclude there is a relationship between mpg and horsepower.

ii) According to the summary table, the R^2 value indicates that about 64.9% of the explained variance
in mpg is due to horsepower.

iii) The t-statistic is negative, so we can conclude there is a negative relationship between mpg and horsepower.

In [58]:
# uses the get_prediction() function
get_prediction()

NameError: name 'get_prediction' is not defined